lib/auth.coffee

```diff
@@ -303,7 +328,8 @@ getAuth = (deps, opts) ->
 	# });
 	###
 	exports.logout = (callback) ->
-		token.remove().asCallback(callback)
+		userDetails = null
```

Comment:
We also need to reset this anywhere you log in, or otherwise change the token. If you're logged in, you check the details, you log in as somebody else, and check the details again, you should get the 2nd login's details, not the 1st one's.

Reply:
@pimterry Would it also make sense to have a timeframe out of which the cache should be invalidated anyway? Eg: 10 minutes?